src/orquestra/opt/optimizers/scipy_optimizer.py

```diff
@@ -66,7 +142,7 @@ def _minimize(
                 evaluations should be recorded.
 
         """
-
+        assert isinstance(cost_function, _CostFunctionWithBestValue)
```

Comment:
Can't we just replace this with type checking in the cost_function input? Not sure why it needs a check here. At most, I would just put an easy to understand error wherever the first conflict arises that requires one to use a `CostFunctionWithBestValue`.

Reply:
I removed this and performed an instance type check instead to be able to deduce which attributes I should access to get the best parameters and function value.